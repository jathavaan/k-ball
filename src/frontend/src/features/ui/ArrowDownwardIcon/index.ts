export * from "./ArrowDownwardIcon.style.ts";
export * from "./ArrowDownwardIcon.types.ts";
export * from "./ArrowDownwardIcon.tsx";
