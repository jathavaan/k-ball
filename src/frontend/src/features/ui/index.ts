export * from "./card-header/CardHeader";
export * from "./card-content-text/CardContentText";
export * from "./error-alert/ErrorAlert";
export * from "./linear-progress-bar/LinearProgressBar";
export * from "./image-container/ImageContainer";
export * from "./select/Select";
