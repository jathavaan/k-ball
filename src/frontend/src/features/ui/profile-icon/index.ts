// index.ts in profileIcon folder
export * from "./profileIcon.api";
export * from "./profileIcon.hooks";
export * from "./profileIcon.types";
