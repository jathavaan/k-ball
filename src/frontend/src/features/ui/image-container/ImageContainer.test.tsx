import { render, screen } from "@testing-library/react";
import { ImageContainer } from "@features/ui";

describe("Tests for ImageContainer", () => {
  it("Should render the image with correct source and alt text", () => {
    const mockProps = {
      src: "test-image.jpg",
      alt: "Test Image",
    };

    render(<ImageContainer {...mockProps} />);

    const imgElement = screen.getByAltText(mockProps.alt);
    expect(imgElement).toBeDefined();
  });

  it("Should match ImageContainer snapshot", () => {
    const mockProps = {
      src: "test-image.jpg",
      alt: "Test Image",
    };

    const { asFragment } = render(<ImageContainer {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
