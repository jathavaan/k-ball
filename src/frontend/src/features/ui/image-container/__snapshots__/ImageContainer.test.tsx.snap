// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for ImageContainer > Should match ImageContainer snapshot 1`] = `
<DocumentFragment>
  <img
    alt="Test Image"
    class="css-1sws5au"
    src="test-image.jpg"
  />
</DocumentFragment>
`;
